portfolio/src/main/webapp/calendarApi.js

```diff
@@ -0,0 +1,292 @@
+        'orderBy': 'startTime'
+      })
+      .then(function(response) {
+        var events = response.result.items;
+
+        // Show message for no imported event if result list is empty.
+        if (events.length == 0) {
+          $('#empty-calendar-import-message').removeClass('d-none');
+          $('#empty-calendar-import-message').show();
+        } else {
+          $('#empty-calendar-import-message').hide(); 
+          for (i = 0; i < events.length; i++) {
+            const event = events[i];
+            const eventName = event.summary;
+
+            // When retrieved from Google Calendar, these time strings
+            // are in the time zone of the user's Calendar's time zone.
+            // These time strings have format YYYY-MM-DDTHH:MM:SS-TimeZoneOffset
+            var startTimeString = event.start.dateTime;
+            if (!startTimeString) {
+              startTimeString = event.start.date;
+            }
+
+            var endTimeString = event.end.dateTime;
+            if (!endTimeString) {
+              endTimeString = event.end.date;
+            }
+
+            const newCalendarEvent = new CalendarEvent(
+                eventName, new Date(startTimeString), new Date(endTimeString));
+
+            // Add these events to the UI under the events list.
+            // On the UI, the event times will be displayed in the user's
+            // current location's time zone.
+            // Each will not get added more than once.
+            const allEvents = collectAllEvents();
+            const doesEventExist = allEvents.reduce(
+                (newEventExists, existingEvent) => newEventExists ||
+                    eventsEqual(newCalendarEvent, existingEvent),
+                /* initialValue= */ false);
+
+            if (!doesEventExist) {
+              updateCalendarEventList(newCalendarEvent);
+            }
+          }
+        }
+      });
+}
+
+/**
+ * Retrieves the logged in user's Google email and uses that email
+ * to embed the user's Google Calendar on the UI.
+ * The calendar is displayed in the user's Google Calendar's time zone.
+ */
+function showCalendarView(user) {
+  const userEmail = encodeURIComponent(user.getBasicProfile().getEmail());
+
+  // Fetches the user's Calendar's time zone. 
+  gapi.client.calendar.settings.get({'setting': 'timezone'})
+      .then((responseTimeZone) => {
+        const userCalendarTimeZone =
+            encodeURIComponent(responseTimeZone.result.value);
+        const calendarViewUrl =
+            'https://calendar.google.com/calendar/embed?src=' + userEmail +
+            '&ctz=' + userCalendarTimeZone + '&mode=WEEK';
+
+        // Updates the UI so that calendar view appears.
+        $('#calendar-view').removeClass('d-none');
+        $('#calendar-view').attr('src', calendarViewUrl);
+        $('#calendar-view').show();
+      });
+}
+
+/**
+ * Asks the user for Write access to the API scope.
+ */
+function addWriteScope() {
```

Comment:
This also seems to never be used.

Reply:
This function is actually what gets triggered by the button click. This change is updated in the HTML file, which I will push in future PRs.